wgpu-types/src/lib.rs

```diff
@@ -5107,13 +5107,16 @@ pub struct SurfaceConfiguration<V> {
     /// AutoNoVsync will gracefully do a designed sets of fallbacks if their primary modes are
     /// unsupported.
     pub present_mode: PresentMode,
-    /// Desired number of buffers in the swap chain.
+    /// Desired maximum number of frames that the presentation engine should queue in advance.
     ///
-    /// Defaults to 3 when created via `wgpu::Surface::get_default_config`.
+    /// Defaults to 2 when created via `wgpu::Surface::get_default_config`.
+    /// Recommended to use 2 (or higher) for high throughput, 1 for low latency.
     ///
-    /// Recommended to use 3 (or higher) for high throughput, 2 for low latency.
     /// This is a hint to the backend implementation and will be clamped to the supported range.
-    pub desired_swap_chain_size: u32,
+    /// As a consequence either the maximum frame latency is set directly on the swap chain,
+    /// or waits on present to avoid exceeding the maximum frame latency,
+    /// or the swap chain size is set to max-latency + 1.
+    pub desired_maximum_frame_latency: u32,
```

Comment:
we should make a note on what happens if the user sets this to zero.

Do all backends clamp this to a valid range?

Reply:
Yeah right now, they all clamp to the range - this is unchanged from the thing we pair programmed on.
I think in the future we want to go as in Connor's original proposal and expose this just like we expose format and size restrictions. We'll then drop the `desired_` part and check again it as with other properties.

Expanding the comment!